superset-frontend/src/SqlLab/components/ResultSet.jsx

```diff
@@ -216,18 +217,45 @@ export default class ResultSet extends React.PureComponent {
         </Alert>
       );
     } else if (query.state === 'success' && query.ctas) {
+      // Get the schema of the temporary table that was created.
+      let schema = query.schema;
+      let tempTable = query.tempTable;
+      // Sync queries only have tempTable in query.results.query
+      if (
+        query.results &&
+        query.results.query &&
+        query.results.query.tempTable
+      ) {
+        tempTable = query.results.query.tempTable;
+      }
+      if (tempTable !== undefined) {
+        const tableNameParts = tempTable.split('.');
+        if (tableNameParts.length > 1) {
+          schema = tableNameParts[0];
+        }
```

Comment:
In reference to earlier discussions, I think it might make sense to emit an error if a user tries to create a temp table that contains a period until we fix this for good.